app/controllers/notes_controller.rb

```diff
@@ -32,30 +42,44 @@ def edit
 
   def update
     note = Note.find(params[:id])
-    if note
-      if note.update(note_params)
-        redirect_to note
+    if has_permission?(note)
+      if note
+        if note.update(note_params)
+          redirect_to note
+        else
+          render :edit
+        end
       else
-        render :edit
+        render html: 'Note not found', status: 404
       end
     else
-      render html: 'Note not found', status: 404
+      flash[:alert] = "You do not have permission to view this page."
+      redirect_to root_path
     end
   end
 
   def destroy
+    if has_permission?
     note = Note.find(params[:id])
-    if note
-      note.destroy
-      flash[:notice] = "Note deleted"
-      redirect_to notes
+      if note
+        note.destroy
+        flash[:notice] = "Note deleted"
+        redirect_to notes
+      else
+        flash[:alert] = note.errors
+      end
     else
-      flash[:alert] = note.errors
+      flash[:alert] = "You do not have permission to delete this note."
+      redirect_to root_path
     end
   end
 
   private
   def note_params
     params.require(:note).permit(:user_id, :category_id, :name, :body, :file_type)
   end
+
+  def has_permission?(notes)
```

Comment:
The argument coming in is notes. Notes is not used in the method call. Also Note.user_id, what photo is that?